### ISSUE TITLE
[Feature] Upgrade to nx 16

### ISSUE THREAD

**Issue by bmvantunes** [nx-dotnet/nx-dotnet#715]
In nx version 16 the packages were renamed to @nx/* 

This package installs @nrwl/js as dependency - in the future we should install @nx/* dependencies.

PS - would you accept a PR upgrading the repo to nx 16? 
I'm asking this because running "npx nx migrate latest" will touch lots of files making that PR gigantic =)

**Comment by AgentEnder** [comment]
Hey @bmvantunes!

There's one main reason that I've not already migrated to Nx 16, and in turn the @nx/js package. @nx/js doesn't exist before v16, and I am putting off bumping the plugin to a new major, although its likely not  a big deal.

The key thing to note is that the plugin **_does_** work with Nx 16, as the smoke tests here: https://github.com/nx-dotnet/nx-dotnet/actions/workflows/smoke.yml always run against the latest release of Nx.

I'm still thinking over exactly how I want to handle this bump, its possible that we can migrate to 16 ourselves and check if NX_VERSION is greater than or less than 16 when deciding what package to install in init.

**Comment by bmvantunes** [comment]
That's fantastic news @AgentEnder =)

I understand your concern about creating a breaking change (version 2.0.0). 

That being said, creating a v2.0.0 of this package has some benefits, like allowing the maintainers to focus on nx >= 16 without fear of breaking changes for nx <= 15 users. Even tho, as you said, maintaining both versions is probably not a huge effort/problem.

During my experience with Angular, I always appreciated the approach taken by ng-bootstrap. They maintained transparency in their documentation, clearly indicating which versions of their library were compatible with Angular and Bootstrap:
<img width="757" alt="Screenshot 2023-05-12 at 19 47 02" src="https://github.com/nx-dotnet/nx-dotnet/assets/9042965/7fa17a9d-4cbf-4d0c-a95b-f64c1970a5a5">

Maybe if we move to v2.0.0 we can include a similar table to help future users and maintainers.

**Comment by AgentEnder** [comment]
Yeah, I'd be happy to have a table similar to that. Should be something like:

<table>
<tr>
 <td>nx-dotnet version
 <td> Supported Nx version range
 <td> Supported .NET version range
</table>

**Comment by jvskriubakken** [comment]
Hi,

I am currently looking over our nx repo to check if it's sound.

When I run` nx report `I get the following warning:

**_The following packages should match the installed version of nx
     - @nrwl/devkit@15.8.5_**

``` 
   Node   : 18.15.0
   OS     : darwin-arm64
   yarn   : 1.22.19
   
   nx                 : 16.6.0
   @nx/js             : 16.6.0
   @nx/jest           : 16.6.0
   @nx/linter         : 16.6.0
   @nx/workspace      : 16.6.0
   @nx/cypress        : 16.6.0
   @nx/devkit         : 16.6.0
   @nrwl/devkit       : 15.8.5
   @nx/eslint-plugin  : 16.6.0
   @nx/plugin         : 16.6.0
   @nx/react          : 16.6.0
   @nx/storybook      : 16.6.0
   @nrwl/tao          : 16.6.0
   @nx/web            : 16.6.0
   @nx/webpack        : 16.6.0
   typescript         : 5.1.6
   ---------------------------------------
   Community plugins:
   @nx-dotnet/core : 1.22.0
   ---------------------------------------
   Local workspace plugins:
         @airmont/nx-plugins
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/devkit@15.8.5
   
   To fix this, run `nx migrate nx@16.6.0`
```

I understand that this happens because nx-dotnet has not yet upgraded it's relationship to nx@16.6.0

I vote for prioritising this feature 🙂
